### PR TITLE
HPCC-23035 Don't trace values written to workunit variables

### PR DESCRIPTION
**Pull request by richardkchapman**
Temporary values stored in Workunit variables may contain sensitive
information - there's no real benefit to outputting them in the tracing and it
can cause some grief.

Note that temporaries are deleted from the workunit at the end of the job and
so cannot be observed in the workunit XML either.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Ran manually, observed logs

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: ecl/eclagent/eclagent.cpp [hunk 1]
```diff
@@ -1119,7 +1119,7 @@ void EclAgent::setResultData(const char * stepname, unsigned sequence, int len,
 
 void EclAgent::doSetResultString(type_t type, const char *name, unsigned sequence, int len, const char *val)
 {
-    LOG(MCsetresult, unknownJob, "setResultString(%s,%d,'%.*s')", nullText(name), sequence, len, val);
+    LOG(MCsetresult, unknownJob, "setResultString(%s,%d,(%d bytes))", nullText(name), sequence, len);
```
Comment (ghalliday) [inline ecl/eclagent/eclagent.cpp]: really len should be unsigned, and this should be %u, but not significant enough to hold this up.

**Comment by ghalliday** [comment]
@richardkchapman please can you recreate this PR - it was referencing the wrong jira in the commit, so I regressed the change (removed the last commit)/

**Comment by richardkchapman** [comment]
@ghalliday See https://github.com/hpcc-systems/HPCC-Platform/pull/13195

**Pull request merged**